core-bundle/src/Command/MigrateCommand.php

```diff
@@ -83,17 +83,35 @@ protected function execute(InputInterface $input, OutputInterface $output): int
     {
         $this->io = new SymfonyStyle($input, $output);
 
-        if (!$input->getOption('dry-run') && !$input->getOption('no-backup') && !$this->backup($input)) {
-            return 1;
-        }
-
-        if ('ndjson' !== $input->getOption('format')) {
-            return $this->executeCommand($input);
-        }
+        $asJson = 'ndjson' === $input->getOption('format');
 
         try {
+            if ($errors = $this->compileConfigurationErrors()) {
+                if ($asJson) {
+                    foreach ($errors as $message) {
+                        $this->writeNdjson('problem', ['message' => $message]);
+                    }
+                } else {
+                    foreach ($errors as $error) {
+                        $this->io->block($error, '!', 'fg=yellow', ' ', true);
+                    }
+
+                    $this->io->error('The database server is not configured properly. Please resolve the above issue(s) and rerun the command.');
+                }
+
+                return 1;
+            }
+
+            if (!$input->getOption('dry-run') && !$input->getOption('no-backup') && !$this->backup($input)) {
+                return 1;
+            }
+
```

Comment:
What code are you referring to? Except for the returns being integers instead of constants, the whole `execute()` function is identical to the one in [Contao 5.0](https://github.com/contao/contao/blob/5.0/core-bundle/src/Command/MigrateCommand.php#L67-L110).

Reply:
True. It seems I was comparing to Contao 4.13. 🤦‍♂️